### ISSUE TITLE
Every action met with "SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data"

### ISSUE THREAD

**Issue by Koko2110** [ubergeek77/Lemmy-Easy-Deploy#25]
Hello. I apologize if this is not the correct place I should seek help for this, but I assume it should be related to the Easy Deploy script.

**Describe the bug**
Ever since the update to v1.2.0 every single action I try taking in my Lemmy instance is met with a red pop up in the bottom left that states "SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data"

I cannot log in, post, or do anything really. Since my instance was rather new, I didn't mind deleting everything and redeploying the installer from scratch, however the issue persists. I didn't manage to find anyone else having encountered anything of the sort either.

Please do let me know if there is any additional info I can or should provide. Thanks a ton in advance!

**Diagnostic Information**
```

==== Docker Information ====
Detected runtime: docker (Docker version 24.0.2, build cb74dfcd85)
Detected compose: docker compose (Docker Compose version 2.19.1)
   Runtime state: OK

==== System Information ====
KERNEL: 6.4.1-arch1-1 (x86_64)
SHELL: /usr/bin/zsh
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
MEMORY:                total        used        free      shared  buff/cache   available
Mem:           7.7Gi       745Mi       6.3Gi        17Mi       869Mi       6.9Gi
Swap:          8.0Gi          0B       8.0Gi

==== Lemmy-Easy-Deploy Information ====
Version: 1.2.3

IMAGE                                CREATED         STATUS
caddy:latest                         4 minutes ago   Up 4 minutes
ghcr.io/ubergeek77/lemmy-ui:0.18.0   4 minutes ago   Up 4 minutes
ghcr.io/ubergeek77/lemmy:0.18.0      4 minutes ago   Up 4 minutes
postgres:15-alpine                   4 minutes ago   Up 4 minutes
asonix/pictrs:0.4.0-beta.19          4 minutes ago   Up 4 minutes

Integrity:
    eeddb2f9f1d800d8210867a49b05a855a65f2181f8e744184465b53840088319  ./deploy.sh
    1e9b0c0988998dcc33cb0fbfdb0e1679229424e724f898b797380adc7d102446  ./templates/Caddy-Dockerfile.template
    c1202e70662dd2228da36a35a0f38ec8fc81bec8964d7315d02e8671a58dd7d7  ./templates/Caddyfile.template
    2537678c7971df36c1ed95f4228d3cfcb15bb4a28a60d939eaf8dd75b5d64a36  ./templates/cloudflare.snip
    c494a610bcb4cd1cfc0a4fe4fb0f6d437b2a84a0ad1625daee240e6dd6f1c910  ./templates/compose-email-volumes.snip
    c9cb4c5fee12930e17798a02ae1bd12e2dc69e149a394c24511bc9d4e6b776d4  ./templates/compose-email.snip
    d9928baea61975bd432e1a4a7439c04468547c9b42e2bb483c6f18d9438665eb  ./templates/docker-compose.yml.template
    a74770730fd0e809d44e5247e1300be04c2b664be13d06bfec00296e3e63861b  ./templates/lemmy-email.snip
    c834cdce9eaf77f38155b404724fdfe66845575386ee516987452aa715642a6f  ./templates/lemmy.hjson.template

==== Settings ====
        CLOUDFLARE: No
 CADDY_DISABLE_TLS: true
   CADDY_HTTP_PORT: 1337
  CADDY_HTTPS_PORT: 443
 LEMMY_TLS_ENABLED: false
      ENABLE_EMAIL: false
         SMTP_PORT: 25
    ENABLE_POSTFIX: false
POSTGRES_POOL_SIZE: 5

==== Generated Files ====
Deploy Version: 0.18.0;0.18.0

total 36K
drwxr-xr-x 2 0 0 4.0K Jul  4 03:19 caddy
-rw-r--r-- 1 0 0   28 Jul  4 03:42 caddy.env
-rw-r--r-- 1 0 0 1.7K Jul  4 03:42 docker-compose.yml
-rw-r--r-- 1 0 0   50 Jul  4 03:19 lemmy.env
-rw-r--r-- 1 0 0  472 Jul  4 03:42 lemmy.hjson
-rw-r--r-- 1 0 0   49 Jul  4 03:19 pictrs.env
-rw-r--r-- 1 0 0   32 Jul  4 03:42 postfix.env
-rw-r--r-- 1 0 0   51 Jul  4 03:19 postgres.env
-rw-r--r-- 1 0 0   14 Jul  4 03:42 version
```

**Comment by ubergeek77** [comment]
I am not sure why this is happening, but if I had to guess, it is because you have changed the HTTP port to `1337`, and set `LEMMY_TLS_ENABLED` to `false`.

Do you still have the issue when these settings are left to their defaults?

**Comment by ubergeek77** [comment]
Well, I am not quite sure why you are trying to use port 1337 or why you have disabled TLS, but I do have a fix for you.

This is happening because Lemmy's backend has not been coded to recognize ports other than 80 or 443. When you send a request on a different port, Lemmy treats it like a different host, and throws a CORS error. The UI tries to read the error as valid JSON code, but since it isn't your operation fails.

To get around this, you need to tell the Lemmy backend to allow the host and port you are trying to use. To do this, you can use the custom environment variables feature I recently added, to set a custom environment variable for the `lemmy` service.

Create this file relative to `deploy.sh`

`./custom/customLemmy.env`

Inside it, put

`LEMMY_CORS_ORIGIN=http://<your-domain>:1337`

Hopefully that works for you! But out of curiosity, what are you trying to do? 😅

**Comment by Koko2110** [comment]
Thank you very much for the fix! I can confirm it works wonderfully now.

What I was trying to do with these unconventional settings was to set up a Lemmy instance for use locally, only on my network for me and a few friends. I changed them according to your suggestion to the person here, who was looking to set up their own instance in a similar way: https://lemmy.ubergeek77.chat/comment/133393

This is why I used port 1337 as well, it's just an easy one to remember and is free on my network. I had no idea the Lemmy backend was not suited for such things. Prior to v1.2.0 and to v.0.18.0 of Lemmy I managed to use it just fine, though, even with these settings. Not sure, but it was obviously a fluke.

Thank you once again for the quick response and for the fix for my weird use case.
